enumerable.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+    i = 0
+    while i < self.length
+      yield(self[i])
+      i += 1     
+    end
+    self
+  end
+  
+  def my_each_with_index
+    return enum_for unless block_given?
+    i =0
+    while i < self.length
+      yield(self[i], i)
+      i += 1   
+    end
+    self
+  end
+
+  def my_select
+    arr= []
+    x = 0
+    while x < self.length
+      arr << self[x] if yield(self[x]) == true
+      x += 1
+    end
+    arr
+  end
+
+  def my_all?
+    value = false
+    if self.empty?
+      value = true
+      return value
+    else
+      i = 0
+      while i < self.length
+        yield(self[i]) || self.empty? ? value = true : value = false
+        i += 1
+      end
+      return value
+    end
+    
+  end
+
+# [].my_all? { |i| i < 4 } # False
+
+  def my_any?
+    value = false
+    if self.empty?
+      return true
+    end
+    i = 0
+    while i < self.length
+      if yield(self[i]) == true
+        return true
+      end
+      i += 1
+    end
+    return value
+  end
+
+  # print [1, 3, 7, 5].my_any { |ele| ele%2 == 0 }
+
+  def my_none?
+    value = true
+    if self.empty?
+      return true
+    end
+    i = 0
+    while i < self.length
+      if yield(self[i]) == true
+        return false
+      end
+      i += 1
+    end
+    return value
+  end
+
+  def my_count(arg = nil)
+    return self.length if arg == nil && !block_given?
+    counter = 0
+    if block_given?
+      i = 0
+      while i < self.length
+        counter += 1 if yield(self[i])
+        i += 1
+      end
+      return counter
+    else
+      i = 0
+      while i < self.length
+        if self[i] == arg
+          counter += 1
+        end
+        i += 1
+      end
+      return counter
+    end
+  end
+
+  def my_inject (acc=nil)
```

Comment:
search =  proc { |memo, word| memo.length > word.length ? memo : word } 
words.my_inject(&search))  is not equal to words.inject(&search)